app/frontend/src/features/communities/CommunityLeadersSection.tsx

```diff
@@ -0,0 +1,92 @@
+import { Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import CircularProgress from "components/CircularProgress";
+import { CommunityLeadersIcon } from "components/Icons";
+import UserSummary from "components/UserSummary";
+import { Community } from "pb/communities_pb";
+import { useState } from "react";
+import makeStyles from "utils/makeStyles";
+
+import CommunityModeratorsDialog from "./CommunityModeratorsDialog";
+import { SectionTitle } from "./CommunityPage";
+import {
+  COMMUNITY_LEADERS,
+  COMMUNITY_LEADERS_DESCRIPTION,
+  COMMUNITY_MODERATORS,
+  NO_MODERATORS,
+  SEE_ALL_MODERATORS,
+} from "./constants";
+import { useListAdmins } from "./hooks";
+
+const useStyles = makeStyles((theme) => ({
+  section: {
+    display: "grid",
+    rowGap: theme.spacing(2),
+  },
+  loadMoreModeratorsButton: {
+    justifySelf: "center",
+  },
+  moderatorsContainer: {
+    display: "grid",
+    gap: theme.spacing(3),
+    gridTemplateColumns: `repeat(auto-fit, minmax(auto, 21.875rem))`,
+  },
+}));
+
+interface CommunityLeadersSectionProps {
+  community: Community.AsObject;
+}
+
+export default function CommunityLeadersSection({
+  community,
+}: CommunityLeadersSectionProps) {
+  const classes = useStyles();
+  const { adminIds, adminUsers, error, isLoading, hasNextPage } = useListAdmins(
+    community.communityId,
+    "summary"
+  );
+  const [isModeratorsDialogOpen, setIsModeratorsDialogOpen] = useState(false);
+
+  return (
+    <>
+      <section className={classes.section}>
+        <SectionTitle icon={<CommunityLeadersIcon />}>
+          {COMMUNITY_LEADERS}
+        </SectionTitle>
+        <Typography variant="body1">{COMMUNITY_LEADERS_DESCRIPTION}</Typography>
+        <Typography variant="h3">{COMMUNITY_MODERATORS}</Typography>
+        {error ? (
+          <Alert severity="error">{error.message}</Alert>
+        ) : isLoading ? (
+          <CircularProgress />
+        ) : adminIds && adminIds.length > 0 ? (
```

Comment:
Time to stop destructuring react-query? Or you prefer this way?

Reply:
I modified the `useListAdmins` hook slightly so in this case it's returning more than the `listAdmins` query object, and since `adminIds` isn't from the main query object, it doesn't make a difference in this case.

But yeah, definitely worth trying out the type narrowing trick we read on the other post more for these kind of pattern more for a bog standard query object!